### PR TITLE
Add support for centos (and others in the RH family)

### PR DESCRIPTION
**Pull request by miguelaferreira**
Provided that the epel yum repo is configured (adding yum-epel to the runlist is enough), this cookbook can now install x11vnc and configure it as a service on CentOS.

### CODE REVIEW

File: templates/default/x11vnc.conf.erb [hunk 1]
```diff
@@ -2,3 +2,4 @@ start on login-session-start
 script
   start-stop-daemon --start --exec /usr/bin/x11vnc -- -xkb -noxrecord -noxfixes -noxdamage -display :0 <%= @password_opt %> -auth /var/run/lightdm/root/:0 -forever -bg -o /var/log/x11vnc.log
 end script
+
```
Comment (jgoulah) [inline templates/default/x11vnc.conf.erb]: why change this file at all? this is a no-op

Comment (miguelaferreira) [inline templates/default/x11vnc.conf.erb]: In your last comment you says to add a new line at the end of every file. That's what I did. Tomorrow I'll for it one more go and will remove changes for the original files.—
Miguel

Comment (jgoulah) [inline templates/default/x11vnc.conf.erb]: not exactly, your original diff was removing the end of line characters which was not great... but I meant only the files you touched, and technically I didn't mean a "new line" or "extra line" at the end of the file but a "line break" at the end of the last line, if that makes sense... (vi editor will do the right thing...)

Comment (miguelaferreira) [inline templates/default/x11vnc.conf.erb]: I've reverted the file to the sate it was before I added the new line

**Comment by miguelaferreira** [comment]
@jgoulah Any chance you can review this pull request? I need to move cookbooks that depend on this one to production and would prefer not to do it with a patched version. Thanks

**Comment by jgoulah** [comment]
I just have to patch mine and make sure nothing breaks, won't be too much longer, sorry for the delay

**Comment by Sauraus** [comment]
What YUM repository are using that you have x11vnc as a package available on CentOS?

AFAIK 'vnc-server' is the official way of installing a VNC server on CentOS. http://wiki.centos.org/HowTos/VNC-Server

File: attributes/default.rb [hunk 2]
```diff
@@ -1,3 +1,14 @@
-default[:x11vnc] = {
-  :password => ""
+default['x11vnc'] = {
+  password: ''
```
Comment (jgoulah) [inline attributes/default.rb:2]: this change is a syntax error on older ruby versions, seems unnecessary even though they are deprecating their support - not sure I see the reason to break backward compatibility?

File: recipes/default.rb [hunk 3]
```diff
@@ -2,27 +2,43 @@
 # Cookbook Name:: x11vnc
 # Recipe:: default
 
-package "x11vnc"
+package 'x11vnc'
 
-service "x11vnc" do
-  action [:enable, :start]
-  provider Chef::Provider::Service::Upstart
```
Comment (jgoulah) [inline recipes/default.rb:9]: turns out debian needs this provider line...so another thing breaking compatibility.  can you set the provider in your platform_family case statement?  Maybe Chef::Provider::Service::Init::Redhat (not sure what works on centos)

**Comment by jgoulah** [comment]
there seem to be a lot of changes bundled up in this one pull request.  It would be better to have a PR (one squashed commit too) with the logic changes and another with all the linting/syntax changes (which I'm not totally convinced that we need given it breaks on 1.8.7, I know they are deprecating that soon but not sure they have quite yet)

File: attributes/default.rb (hunk 2 above)
Comment (miguelaferreira) [inline attributes/default.rb:2]: I see the point for backwards compatibility if you need it. However this syntax is Ruby 1.9 and that is not even the latest version. I can rever this change.

File: recipes/default.rb (hunk 3 above)
Comment (miguelaferreira) [inline recipes/default.rb:9]: I can fix that too.

**Comment by miguelaferreira** [comment]
@Sauraus: I'm suing EPEL6

**Comment by miguelaferreira** [comment]
@jgoulah: I can address the compatibility issues you found, but I would prefer not to go through all the changes again to separate the linting from the rest. Please let me know if that is a problem for you.

**Comment by jgoulah** [comment]
@miguelaferreira its fine this go round, in the future it would be good to have more clear pull requests (one major change per req)

the provider line is the only breaking change (on debian), the syntax "fixes" would probably be better to defer to the near future when 1.8.7 support is dropped

**Comment by Sauraus** [comment]
@miguelaferreira that's not a standard CentOS yum repo AFAIK, your code does not work on a plain-vanilla CentOS 6.5 install.

Besides why not just use the vnc-server package on CentOS instead? It comes from the standard repos.

**Comment by miguelaferreira** [comment]
I will close this PR to break it up and will make two PRs shortly after.

**Pull request closed without merging**